pkg/util/dynamichelper/dynamichelper.go

```diff
@@ -438,65 +439,83 @@ func mergeGK(old, new kruntime.Object) (kruntime.Object, bool, string, error) {
 	changed := false
 	switch new.(type) {
 	case *appsv1.Deployment:
-		new, expect := new.(*appsv1.Deployment), expect.(*appsv1.Deployment)
-		for i := range expect.Spec.Template.Spec.Containers {
-			ec := &expect.Spec.Template.Spec.Containers[i]
-			nc := &new.Spec.Template.Spec.Containers[i]
-			if ec.Image != nc.Image {
-				ec.Image = nc.Image
-				changed = true
-			}
-			if cmpAndCopy(&nc.Resources.Limits, &ec.Resources.Limits) {
-				changed = true
-			}
-			if cmpAndCopy(&nc.Resources.Requests, &ec.Resources.Requests) {
-				changed = true
-			}
-		}
+		changed = handleDeployment(new, expect)
 	case *admissionregistrationv1.ValidatingWebhookConfiguration:
-		new, expect := new.(*admissionregistrationv1.ValidatingWebhookConfiguration), expect.(*admissionregistrationv1.ValidatingWebhookConfiguration)
-		for i := range expect.Webhooks {
-			if expect.Webhooks[i].FailurePolicy != nil &&
-				new.Webhooks[i].FailurePolicy != nil &&
-				*expect.Webhooks[i].FailurePolicy != *new.Webhooks[i].FailurePolicy {
-				*expect.Webhooks[i].FailurePolicy = *new.Webhooks[i].FailurePolicy
-				changed = true
-			}
-			if expect.Webhooks[i].TimeoutSeconds != nil &&
-				new.Webhooks[i].TimeoutSeconds != nil &&
-				*expect.Webhooks[i].TimeoutSeconds != *new.Webhooks[i].TimeoutSeconds {
-				*expect.Webhooks[i].TimeoutSeconds = *new.Webhooks[i].TimeoutSeconds
-				changed = true
-			}
-		}
+		changed = handleValidatingWebhook(new, expect)
 	case *admissionregistrationv1.MutatingWebhookConfiguration:
-		new, expect := new.(*admissionregistrationv1.MutatingWebhookConfiguration), expect.(*admissionregistrationv1.MutatingWebhookConfiguration)
-		for i := range expect.Webhooks {
-			if expect.Webhooks[i].FailurePolicy != nil &&
-				new.Webhooks[i].FailurePolicy != nil &&
-				*expect.Webhooks[i].FailurePolicy != *new.Webhooks[i].FailurePolicy {
-				*expect.Webhooks[i].FailurePolicy = *new.Webhooks[i].FailurePolicy
-				changed = true
-			}
-			if expect.Webhooks[i].TimeoutSeconds != nil &&
-				new.Webhooks[i].TimeoutSeconds != nil &&
-				*expect.Webhooks[i].TimeoutSeconds != *new.Webhooks[i].TimeoutSeconds {
-				*expect.Webhooks[i].TimeoutSeconds = *new.Webhooks[i].TimeoutSeconds
-				changed = true
-			}
-		}
+		changed = handleMutatingWebhook(new, expect)
 	}
-
 	var diff string
 	if _, ok := expect.(*corev1.Secret); !ok { // Don't show a diff if kind is Secret
 		diff = cmp.Diff(old, expect)
 	}
-
 	return expect, changed, diff, nil
 }
 
-func cmpAndCopy(srcP, dstP *corev1.ResourceList) bool {
-	src, dst := *srcP, *dstP
+func handleDeployment(new, expect kruntime.Object) bool {
+	changed := false
+	newDeployment, expectDeployment := new.(*appsv1.Deployment), expect.(*appsv1.Deployment)
```

Comment:
```suggestion
	newDeployment, expectedDeployment := new.(*appsv1.Deployment), expected.(*appsv1.Deployment)
```

Reply:
btw, shouldn't we be checking if the type casting is successful? otherwise we might end up with panics? (here, but also in the webhook cases)

Reply:
the cast cannot fail as it is checked just before the func call:
`	case *appsv1.Deployment:
		changed = handleDeployment(new, expected)`